internal/cephfs/fsjournal.go

```diff
@@ -97,6 +120,25 @@ func checkVolExists(ctx context.Context, volOptions *volumeOptions, secret map[s
 	util.DebugLog(ctx, "Found existing volume (%s) with subvolume name (%s) for request (%s)",
 		vid.VolumeID, vid.FsSubvolName, volOptions.RequestName)
 
+	if sID != nil {
+		// This is a work around to fix sizing issue for cloned images
+		err = resizeVolume(ctx, volOptions, cr, volumeID(vid.FsSubvolName), volOptions.Size)
+		if err != nil {
+			klog.Errorf(util.Log(ctx, "failed to expand volume %s: %v"), volumeID(vid.FsSubvolName), err)
+			return nil, err
+		}
+	}
+	if parentVolOpt != nil {
+		err = checkCloneFromSubvolumeExists(ctx, volumeID(pvID.FsSubvolName), volumeID(vid.FsSubvolName), volOptions, parentVolOpt, cr)
```

Comment:
Clone existence checks are already performed above. Rename this function as `cleanupCloneFromSubvolumeSnapshot`.

Also move the resize outside the `checkCloneFromSubvolumeExists` and doing it common for both types of clone